### DISCUSSION TITLE
Does hidden step_factor make my simulation longer?

### DISCUSSION THREAD

**Discussion by kevinPoliPoli** [OFFIS-mosaik/mosaik#9] [General]
Hi everyone, I'm trying to use Mosaik with an FMU. Since then I have used the repo [Mosaik-fmi](https://gitlab.com/mosaik/components/mosaik-fmi/-/tree/master?ref_type=heads) but I recently discovered this [branch](https://gitlab.com/mosaik/components/mosaik-fmi/-/tree/import_fmi3_from_rep?ref_type=heads). I compared the master branch implementation with the last one and I obtain a strange behavior: specifically my FMU simulates an electric vehicle and outputs the level of charge after walking through a path. Comparing the same model inputs, using the last version of the linked branch the model outputs a significantly lower residual charge. Moreover, the simulation takes longer, in the first case takes 4 seconds to complete, and in the second one 57! So, maybe there is some hidden initialization parameter like the one in the main branch named step_factor which simulates more time? 

These are the initialization parameters:
### first repo
`
fmu_sim = world.start('FMI', step_size=1, step_factor=1,
                          stop_time=100000, work_dir=os.path.join(os.getcwd(), 'src/fmu/eGolf'), fmu_name='eGolf',
                          fmi_version='2', fmi_type='cs', logging_on=False, instance_name='eGolf')  
`
### second repo
`
fmu_sim = world.start('FMI3', step_size=1,
                          stop_time=100000, work_dir='/src/fmu/eGolf', fmu_name='eGolf',
                          fmi_version='2', fmi_type='cs', logging_on=False, instance_name='eGolf',
                          show_result=False, print_model=False, log_input=True)
`

**Comment by VaanArima** [comment]
Hello, I am the research assistant working on this branch. The problem you describe is a known bug related to the way model inputs are processed with the new FMPy library. I am currently investigating this problem. With the switch to FMPy, the step_factor parameter is no longer needed and has therefore been removed. I suspect that the increased solution time is an independent problem. I thank you for reporting these bugs and will work on solving them as soon as possible. Is the FMU you used openly available, so I could look at it myself?
